packages/angular/src/generators/setup-mfe/lib/add-remote-to-host.ts

```diff
@@ -17,61 +21,106 @@ export function checkIsCommaNeeded(mfeRemoteText: string) {
     : false;
 }
 
-export function addRemoteToHost(host: Tree, options: Schema) {
+export function addRemoteToHost(tree: Tree, options: Schema) {
   if (options.mfeType === 'remote' && options.host) {
-    const hostProject = readProjectConfiguration(host, options.host);
-    const hostMfeConfigPath = joinPathFragments(
-      hostProject.root,
-      'mfe.config.js'
+    const hostProject = readProjectConfiguration(tree, options.host);
+    const pathToMfeManifest = joinPathFragments(
+      hostProject.sourceRoot,
+      'assets/mfe.manifest.json'
+    );
+    const hostFederationType = determineHostFederationType(
+      tree,
+      pathToMfeManifest
     );
 
-    if (!hostMfeConfigPath || !host.exists(hostMfeConfigPath)) {
-      throw new Error(
-        `The selected host application, ${options.host}, does not contain a mfe.config.js. Are you sure it has been set up as a host application?`
-      );
+    if (hostFederationType === 'static') {
+      addRemoteToStaticHost(tree, options, hostProject);
+    } else if (hostFederationType === 'dynamic') {
+      addRemoteToDynamicHost(tree, options, pathToMfeManifest);
     }
 
-    const hostMFEConfig = host.read(hostMfeConfigPath, 'utf-8');
-    const webpackAst = tsquery.ast(hostMFEConfig);
-    const mfRemotesNode = tsquery(
-      webpackAst,
-      'Identifier[name=remotes] ~ ArrayLiteralExpression',
-      { visitAllChildren: true }
-    )[0] as ArrayLiteralExpression;
-
-    const endOfPropertiesPos = mfRemotesNode.getEnd() - 1;
-    const isCommaNeeded = checkIsCommaNeeded(mfRemotesNode.getText());
-
-    const updatedConfig = `${hostMFEConfig.slice(0, endOfPropertiesPos)}${
-      isCommaNeeded ? ',' : ''
-    }'${options.appName}',${hostMFEConfig.slice(endOfPropertiesPos)}`;
-
-    host.write(hostMfeConfigPath, updatedConfig);
-
     const declarationFilePath = joinPathFragments(
       hostProject.sourceRoot,
       'decl.d.ts'
     );
 
     const declarationFileContent =
-      (host.exists(declarationFilePath)
-        ? host.read(declarationFilePath, 'utf-8')
+      (tree.exists(declarationFilePath)
+        ? tree.read(declarationFilePath, 'utf-8')
         : '') + `\ndeclare module '${options.appName}/Module';`;
-    host.write(declarationFilePath, declarationFileContent);
+    tree.write(declarationFilePath, declarationFileContent);
 
-    addLazyLoadedRouteToHostAppModule(host, options);
+    addLazyLoadedRouteToHostAppModule(tree, options, hostFederationType);
   }
 }
 
+function determineHostFederationType(
+  tree: Tree,
+  pathToMfeManifest: string
+): 'dynamic' | 'static' {
+  return tree.exists(pathToMfeManifest) ? 'dynamic' : 'static';
+}
+
+function addRemoteToStaticHost(
+  tree: Tree,
+  options: Schema,
+  hostProject: ProjectConfiguration
+) {
+  const hostMfeConfigPath = joinPathFragments(
+    hostProject.root,
+    'mfe.config.js'
+  );
+
+  if (!hostMfeConfigPath || !tree.exists(hostMfeConfigPath)) {
+    throw new Error(
```

Comment:
Maybe mention `mfe.manfest.json` as well in case the user meant to use dynamic host -- maybe they renamed the file without realizing the problem.